documentation/src/main/asciidoc/cassandraclient.adoc

```diff
@@ -0,0 +1,303 @@
+
+All the information between the browser and the server is formatted as JSON.
+
+The elements are stored in the Cassandra Database.
+
+
+== Solution
+
+We recommend that you follow the instructions in the next sections and create the application step by step.
+However, you can go right to the completed example.
+
+The solution is located in the `quickstart` {quickstart-clone-url}[directory].
+
+== Creating the Maven project
+
+First, we need a new project. Create a new maven project and copy the `pom.xml` file that is present in the `quickstart`.
+
+The `pom.xml` is importing the RESTEasy/JAX-RS, JSON-B, Context Propagation and Cassandra Client extensions.
+We will be building a REST application that is using DataStax Object Mapper. It will simplify the Data Access Layer code.
+
+The most important part of the `pom.xml` is adding the `cassandra-quarkus` extension:
+[source,xml]
+----
+<dependency>
+  <groupId>com.datastax.oss.quarkus</groupId>
+  <artifactId>cassandra-quarkus-client</artifactId>
+  <version>${quarkus.version}</version>
+</dependency>
+----
+
+== Creating JSON REST service
+In this example, we will create an application to manage a list of fruits.
+
+First, let's create the `Fruit` bean as follows:
+[source,java]
+----
+@Entity
+public class Fruit {
+
+  @ClusteringColumn private String name;
+  private String description;
+  @PartitionKey private String id;
+
+  public Fruit() {}
+
+  public Fruit(String name, String description, String id) {
+    this.name = name;
+    this.description = description;
+    this.id = id;
+  }
+  // getters, setters, hashCode and equals omitted for brevity
+}
+----
+We are using ObjectMapper, due to that fact the class is annotated with an `@Entity`.
+Also, the Fruit `id` is a `@PartitionKey` and `name` is a `@ClusteringColumn` so we are using the corresponding annotations
+from the ObjectMapper. It will allow the Mapper to generate proper CQL queries underneath.
+
+One important thing to note is that having a default constructor is required by the Object Mapper.
+
+To leverage the Mapper logic in this app we need to create a DAO:
+[source,java]
+----
+@Dao
+public interface FruitDao {
+  @Update
+  void update(Fruit fruit);
+
+  @Select
+  PagingIterable<Fruit> findById(String id);
+}
+----
+
+This class exposes operations that will be used in the REST service.
+
+Finally, the Mapper itself:
+[source,java]
+----
+@Mapper
+public interface FruitMapper {
+  @DaoFactory
+  FruitDao fruitDao(@DaoKeyspace CqlIdentifier keyspace);
+}
+----
+It is constructing the `FruitDao`.
+
+Now create a `com.datastax.oss.quarkus.FruitService` that will be the business layer of our application and store/load the fruits from the Cassandra database.
+[source,java]
+----
+@ApplicationScoped
+public class FruitService {
+  private final FruitDao dao;
+
+  @Inject
+  public FruitService(CqlSession session) {
+    dao = new FruitMapperBuilder(session).build().fruitDao(CqlIdentifier.fromCql("k1"));
```

Comment:
1. I can create an application bean that is `FruitDao`, but I am not sure if it is needed. It will add some complexity to this example by adding one additional layer, wdyt?

2. Regarding using the session name from the config file, I am not sure if we should provide it.
- We can add one additional setting in the application.properties of our starter called keyspace, but it will restrict usage of the `CqlSession` to one keyspace. I think it may be too limiting.
- Right now, the problem of providing keyspace name is on the user - he needs to use any mechanism that he wants, to propagate it. The user when creating own quarkus app can use their own `@ConfigRoot` and propagate it, or have it hardcoded. 

I am not sure which direction is better, for spring we did configure keyspace at the start and restricted the `CqlSession` to one keyspace. Maybe we should wait for feedback before changing it?

Reply:
Keyspace: right now it's hardcoded, so it's even less flexible. I think moving it to the configuration is good enough for this example. The only case it doesn't cover is having two `FruitService` beans that hit different keyspaces: that's possible, but doesn't seem like a very common use case.

Injecting `FruitDao`: I'm not sure if I would go that far, but we should at least inject the `FruitMapper`. A typical application can probably have more services (`StoreService`, `UserService`, etc), and in such a case you don't want to re-create a mapper instance for every service.

Reply:
1. Keyspace from settings implemented in https://github.com/datastax/cassandra-quarkus/pull/9/commits/97821a137d3fee2823388eef83979271e242e30a and docs 
https://github.com/datastax/cassandra-quarkus/pull/9/commits/76a99f9f0f2ecb945b50a28ac70af674f3e606a3

2. Injecting `FrutiMapper`: implemented in https://github.com/datastax/cassandra-quarkus/pull/9/commits/ab57a565a7d5fd85e82d81a41a39c19d20b4720b and docs
https://github.com/datastax/cassandra-quarkus/pull/9/commits/cb0be06dc99f4762bcbaee47da3e9337deb9ae0e